backend/projects/serializers.py

```diff
@@ -143,6 +145,20 @@ def _has_siblings(self, obj) -> bool:
             > 1
         )
 
+    def _user_metrics(self, obj):
+        metrics = list(
+            MetricsPreferences.objects.filter(project=obj).filter(
+                user_id=self.context.get("user_id")
+            )
+        )
+        if len(metrics) == 0:
```

Comment:
Are that the default metrics? Shouldn’t we set that as a default in the DB migration?

Reply:
If we set these metrics as default in the DB migration, we need to insert a line for each user and for each of his/her project in the table. So I don't think this is ideal, what do you think ?

Reply:
I second @MathildeDuboille, this will save a lot of space in the database

Reply:
Hmm I see your point. However I think that the typical Falco instance has few users and few projects. Even for Theodo’s Falco (which I think would be one of the biggest one), we have 43 projects and ~130 user, and every user averages 1-2 projects. So that would mean <300 lines in DB, which I think is definitely manageable.

I might be mistaken, but wouldn’t using the DB default to store the three defaults simplify the mechanism for creating / updating the metrics ? For one, creation would be already taken care of. Also, there’s something “weird” about asking for an object in an API, receiving one, and not being able to tell if 1) that’s the state of the DB 2) the DB’s empty and we’re receiving default.

Reply:
Also I should note that I do not consider this topic as blocking, and we can totally merge this and have this discussion in a later issue. Just think it’s an interesting discussion!

Reply:
FYI, the free DB instance in Heroku is capped at 10,000 lines, and the cheapest one (the one we have at Theodo) is capped at 10,000,000 lines.
So either you do not pay and will likely have only a few number of projects (otherwise you’ll reach the max pretty quickly), or you do pay and a few hundred lines are quite insignificant.

Reply:
Ok that's convincing, we should put these default metrics in the database